lib/cadet/assessments/submission.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Cadet.Assessments.Submission do
+  @moduledoc false
+  use Cadet, :model
+
+  import Ecto.Query
+
+  alias Cadet.Assessments.Assessment
+  alias Cadet.Assessments.SubmissionStatus
+  alias Cadet.Accounts.User
+  alias Cadet.Assessments.Mission
+  alias Cadet.Assessments.Answer
+
+  schema "submissions" do
+    field(:status, SubmissionStatus, default: :attempting)
+    field(:submitted_at, Timex.Ecto.DateTime)
+    field(:override_xp, :integer)
+
+    belongs_to(:mission, Mission)
+    belongs_to(:student, User)
+    belongs_to(:grader, User)
+    has_many(:answers, Answer)
+
+    timestamps()
+  end
+
+  @required_fields ~w(status)a
+  @optional_fields ~w(override_xp submitted_at)a
+
+  def changeset(submission, params) do
+    params = convert_date(params, :submitted_at)
+
+    submission
+    |> cast(params, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
+    |> validate_role(:student, "student")
+    |> validate_role(:grader, "staff")
+  end
+
+  def validate_role(changeset, user, role) do
+    validate_change(changeset, user, fn ^user, user ->
+      case user.role == role do
+        true -> []
+        false -> [{user, "Access Denied"}]
```

Comment:
change error message to `"does not have the role #{role}"`.

Rule about validation error message is that they will be appended to the field name, so, e.g. they should be all lowercase.

In this case, following your code, the error message will be e.g. `Student Access Denied` which does not make sense as compared to `student does not have the role student`